### PR TITLE
Added the `OnSignRead` handler and `SignRead` event, and prevent out of bounds tile access

### PR DESCRIPTION
**Pull request by drunderscore**
Added check to ensure the sign being read is within world bounds (`x >= 0
&& y >= 0 && x < Main.maxTilesX && y < Main.maxTilesY`)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ This is the rolling changelog for TShock for Terraria. Use past tense when addin
   * Do not forget to sign every line you change with your name. (@hakusaro)
   * If there is no section called "Upcoming changes" below this line, please add one with `## Upcoming changes` as the first line, and then a bulleted item directly after with the first change.
 
+## Upcoming changes
+* Added the `OnSignRead` handler in `GetDataHandler`, and added the `SignRead` event. Added check to ensure the sign being read is within world bounds (x >= 0 && y >= 0 && x < Main.maxTilesX && y < Main.maxTilesY). (@drunderscore)
```
Comment (hakusaro) [inline CHANGELOG.md]: Can you add this to the .12 section below? Same issue as the other PR, unfortunately. Not a blocker for merge though.

Comment (hakusaro) [reply CHANGELOG.md]:
```suggestion
* Added the `OnSignRead` handler in `GetDataHandler`, and added the `SignRead` event. Added check to ensure the sign being read is within world bounds `(x >= 0 && y >= 0 && x < Main.maxTilesX && y < Main.maxTilesY)`. (@drunderscore)
```

File: TShockAPI/GetDataHandlers.cs [hunk 2]
```diff
@@ -1187,6 +1188,40 @@ private static bool OnPlayerTeam(TSPlayer player, MemoryStream data, byte _id, b
...
+		/// <summary>
+		/// For use in a SignRead event
+		/// </summary>
+		public class SignReadEventArgs : GetDataHandledEventArgs
+		{
+			/// <summary>
+			/// X location of the sign
+			/// </summary>
+			public int X { get; set; }
+			/// <summary>
```
Comment (hakusaro) [inline TShockAPI/GetDataHandlers.cs:1201]: Can you add linebreaks in reasonable spots after things here?

**Review by hakusaro — DISMISSED** [review]
lgtm, minor nits, that's all
lgtm_trick

**Comment by drunderscore** [comment]
Sorry about the double CI run :(
all requests changes implemented

**Comment by hakusaro** [comment]
> Sorry about the double CI run :(
all requests changes implemented

We pay $0 for CI so it’s k

**Review by hakusaro — APPROVED** [review]
lgtm_trick

**Pull request merged**